### PR TITLE
docs: update docs related to the changes introduced by generics

### PR DESCRIPTION
**Pull request by digorithm**

### CODE REVIEW

**Review by Salka1988 — APPROVED** [review]
LGTM :+1:

**Review by Br1ght0ne — DISMISSED** [review]

File: docs/src/types/bits256.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Bits256
+
+In Fuel, a type called `b256` is used to represent hashes and holds a 256-bit value. The Rust SDK represents `b256` as `Bits256(value)` where `value` is a `[u8; 32]`. If your contract method takes a `b256` as input, you must pass a `Bits256([u8; 32])` when calling it from the SDK.
```
Comment (MujkicA) [inline docs/src/types/bits256.md:3]:
```suggestion
In Fuel, a type called `b256` represents hashes and holds a 256-bit value. The Rust SDK represents `b256` as `Bits256(value)` where `value` is a `[u8; 32]`. If your contract method takes a `b256` as input, you must pass a `Bits256([u8; 32])` when calling it from the SDK.
```

File: docs/src/types/bytes32.md [hunk 2]
```diff
@@ -10,4 +10,4 @@ These are the main ways of creating a `Bytes32`:
 
 However, there are more ways to achieve that and `Bytes32` implements many more useful traits, see the [fuel-types documentation](https://docs.rs/fuel-types/latest/fuel_types/struct.Bytes32.html).
 
-> **Note:** In Sway, there's a special type called `b256`, which is similar to `Bytes32`; also used to represent hashes and it holds a 256-bit value. In Rust, through the SDK, this is also represented as `[u8; 32]`. If your contract method takes a `b256` as input, all you need to do is pass a `[u8; 32]` when calling it from the SDK.
+> **Note:** In Fuel, there's a special type called `b256`, which is similar to `Bytes32`; also used to represent hashes and it holds a 256-bit value. In Rust, through the SDK, this is represented as `Bits256(value)` where `value` is a `[u8; 32]`. If your contract method takes a `b256` as input, all you need to do is pass a `Bits256([u8; 32])` when calling it from the SDK.
```
Comment (MujkicA) [inline docs/src/types/bytes32.md:13]:
```suggestion
> **Note:** In Fuel, there's a special type called `b256`, which is similar to `Bytes32`; also used to represent hashes, and it holds a 256-bit value. In Rust, through the SDK, this is represented as `Bits256(value)` where `value` is a `[u8; 32]`. If your contract method takes a `b256` as input, all you need to do is pass a `Bits256([u8; 32])` when calling it from the SDK.
```

**Review by MujkicA — APPROVED** [review]
tiny nits

**Review by segfault-magnet — APPROVED** [review]
Ah.. I should have done this in the generics PR, sorry and ty

**Review by hal3e — DISMISSED** [review]

**Review by Br1ght0ne — APPROVED** [review]
:shipit:

**Pull request merged**